### PR TITLE
Sort groups by name and creation time.

### PR DESCRIPTION
**Pull request by houcros**

### CODE REVIEW

File: src/test/scala/com/advancedtelematic/ota/deviceregistry/data/GroupGenerators.scala [hunk 1]
```diff
@@ -19,16 +19,14 @@ trait GroupGenerators {
 
   val genGroupName: Gen[Group.Name] = for {
     strLen <- Gen.choose(2, 100)
-    name   <- Gen.listOfN[Char](strLen, Arbitrary.arbChar.arbitrary)
+    name   <- Gen.listOfN[Char](strLen, Gen.alphaNumChar)
```
Comment (houcros) [inline src/test/scala/com/advancedtelematic/ota/deviceregistry/data/GroupGenerators.scala]: I had to change this if we were to make any sense of sorting by group name. Hopefully this doesn't cause any regression (?)

Comment (simao) [reply src/test/scala/com/advancedtelematic/ota/deviceregistry/data/GroupGenerators.scala]: what do you mean? Why did you have to change it ?

Comment (houcros) [reply src/test/scala/com/advancedtelematic/ota/deviceregistry/data/GroupGenerators.scala]: We were generating group names with any Unicode chars. It's unclear to me how to assert they're sorted, hence very awkward to test. I don't think sorting by code-point is what we want.

Comment (simao) [reply src/test/scala/com/advancedtelematic/ota/deviceregistry/data/GroupGenerators.scala]: We do want to keep testing with unicode chars though. Maybe you could use a different generator for your test...

Comment (houcros) [reply src/test/scala/com/advancedtelematic/ota/deviceregistry/data/GroupGenerators.scala]: Yes, that was indeed my first approach. But that would imply that we accept any chars for the name, but sort only over ASCII chars right?
If we're fine with that I will just use another group name generator.

Comment (simao) [reply src/test/scala/com/advancedtelematic/ota/deviceregistry/data/GroupGenerators.scala]: that is not great but it's better than not testing with unicode at all

Comment (koshelev) [reply src/test/scala/com/advancedtelematic/ota/deviceregistry/data/GroupGenerators.scala]: If i understand it correctly, our API allows any valid UTF-8 character as an input. So I guess the test should have the same constraint

Comment (houcros) [reply src/test/scala/com/advancedtelematic/ota/deviceregistry/data/GroupGenerators.scala]: So: I added a new group name generator that it's used only for the sorting test.

File: src/test/scala/com/advancedtelematic/ota/deviceregistry/data/GroupGenerators.scala [hunk 2]
```diff
@@ -22,13 +22,16 @@ trait GroupGenerators {
     name   <- Gen.listOfN[Char](strLen, Arbitrary.arbChar.arbitrary)
   } yield Refined.unsafeApply(name.mkString)
 
-  def genGroupInfo: Gen[Group] =
-    for {
-      name <- genGroupName
-    } yield Group(GroupId.generate(), name, defaultNs, GroupType.static, None)
+  val genAlphaNumGroupName: Gen[Group.Name] = for {
```
Comment (simao) [inline src/test/scala/com/advancedtelematic/ota/deviceregistry/data/GroupGenerators.scala]: I think would be better to have only one generator that accepts a Gen[Char] with `Arbitrary.arbChar.arbitrary` as the default, then you can pass `Gen.alphaNumChar` on your test

**Comment by simao** [comment]
lgtm

**Pull request merged**